Functional.Primitives/Helpers.cs

```diff
@@ -6,4 +6,27 @@ namespace Functional
 {
 	[AttributeUsage(AttributeTargets.Class | AttributeTargets.Constructor | AttributeTargets.Method, AllowMultiple = false)]
 	internal class AllowAllocationsAttribute : Attribute { }
+
+	internal static class DelegateCache<T>
+	{
+		public static readonly Func<T, T> Passthrough = x => x;
+	}
+
+	internal static class DelegateCache<TIn, TOut>
+	{
+		private static readonly Exception _expectedSuccessException = new InvalidOperationException("Not a successful result!");
+		private static readonly Exception _expectedFailureException = new InvalidOperationException("Not a faulted result!");
+
+		public static readonly Func<TIn, TOut> ExpectedSuccess = _ => throw _expectedSuccessException;
```

Comment:
Will this not cause a mutation on the stack trace of the static `_expectedSuccessException`?

Reply:
It may... I am less worried about it here since the method is `internal` to this assembly and used in a very specific spot. That's not necessarily a good reason to keep it this way though.

Reply:
The exceptions will trickle up the stack and hit external code though, no? I mean, it probably won't be a problem, unless... hmm, threading, yea it could be a problem there.

I'm guessing this is an optimization to not have to allocate memory for the exceptions? If so, I'd personally be not so concerned with performance optimization on exceptional paths like this, but if there's something I'm missing I'd be interested to know.

Reply:
The delegate caching is not needed. It's an artifact of attempting to write an allocation-free `Zip` so I will remove it.

Reply:
Removed!